selftests/unit/test_archive.py

```diff
@@ -155,6 +155,14 @@ def test_empty_tbz2(self):
         self.assertEqual(ret, None, "Empty archive should return None (%s)"
                          % ret)
 
+    def test_gzip(self):
```

Comment:
Why not test the new functions directly and with more granularity? So we should have:

```python
def test_is_gzip_file()
def test_gzip_uncompress()
def test_is_archive_gzip()
def test_uncumpress_gzip()
```

Reply:
Yep, good point.